### DISCUSSION TITLE
promql have no data

### DISCUSSION THREAD

**Discussion by unihon** [apache/skywalking#11660] [Q&A]
version: 9.4.0

Why do some promql queries have no data?

is ok:
```
endpoint_resp_time{service='my-service', layer='GENERAL' ,endpoint="/api/xx"}
```

is no:
```
endpoint_resp_time{}
endpoint_resp_time{service='my-service', layer='GENERAL'}
endpoint_resp_time{service='my-service', layer='GENERAL' ,endpoint=~"/api/xx|/api/yy"}
```

**Comment by wu-sheng** [comment]
Why do you expect to have data if you don't provide a complete context for an endpoint? You need a service, layer and endpoint name to build the query. All fields are required.
